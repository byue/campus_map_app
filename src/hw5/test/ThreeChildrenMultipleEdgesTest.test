# Tests that a linear graph of 3 nodes can be created,
# and that the three nodes can have multiple connections
# with different labels.

# Create a graph
CreateGraph graph1

# Add 3 nodes
AddNode graph1 n1
AddNode graph1 n2
AddNode graph1 n3

# Connect nodes with multiple connections with label a
AddEdge graph1 n1 n1 a
AddEdge graph1 n1 n2 a
AddEdge graph1 n1 n3 a
AddEdge graph1 n2 n1 a
AddEdge graph1 n2 n2 a
AddEdge graph1 n2 n3 a
AddEdge graph1 n3 n1 a
AddEdge graph1 n3 n2 a
AddEdge graph1 n3 n3 a

# Connect nodes with multiple connections with label c
AddEdge graph1 n1 n1 c
AddEdge graph1 n1 n2 c
AddEdge graph1 n1 n3 c
AddEdge graph1 n2 n1 c
AddEdge graph1 n2 n2 c
AddEdge graph1 n2 n3 c
AddEdge graph1 n3 n1 c
AddEdge graph1 n3 n2 c
AddEdge graph1 n3 n3 c

# Connect nodes with multiple connections with label b
AddEdge graph1 n1 n1 b
AddEdge graph1 n1 n2 b
AddEdge graph1 n1 n3 b
AddEdge graph1 n2 n1 b
AddEdge graph1 n2 n2 b
AddEdge graph1 n2 n3 b
AddEdge graph1 n3 n1 b
AddEdge graph1 n3 n2 b
AddEdge graph1 n3 n3 b

# list nodes of graph
ListNodes graph1

# list children of each node
ListChildren graph1 n1
ListChildren graph1 n2
ListChildren graph1 n3