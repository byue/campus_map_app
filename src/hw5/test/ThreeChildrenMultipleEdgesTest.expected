# Tests that a linear graph of 3 nodes can be created,
# and that the three nodes can have multiple connections
# with different labels.

# Create a graph
created graph graph1

# Add 3 nodes
added node n1 to graph1
added node n2 to graph1
added node n3 to graph1

# Connect nodes with multiple connections with label a
added edge a from n1 to n1 in graph1
added edge a from n1 to n2 in graph1
added edge a from n1 to n3 in graph1
added edge a from n2 to n1 in graph1
added edge a from n2 to n2 in graph1
added edge a from n2 to n3 in graph1
added edge a from n3 to n1 in graph1
added edge a from n3 to n2 in graph1
added edge a from n3 to n3 in graph1

# Connect nodes with multiple connections with label c
added edge c from n1 to n1 in graph1
added edge c from n1 to n2 in graph1
added edge c from n1 to n3 in graph1
added edge c from n2 to n1 in graph1
added edge c from n2 to n2 in graph1
added edge c from n2 to n3 in graph1
added edge c from n3 to n1 in graph1
added edge c from n3 to n2 in graph1
added edge c from n3 to n3 in graph1

# Connect nodes with multiple connections with label b
added edge b from n1 to n1 in graph1
added edge b from n1 to n2 in graph1
added edge b from n1 to n3 in graph1
added edge b from n2 to n1 in graph1
added edge b from n2 to n2 in graph1
added edge b from n2 to n3 in graph1
added edge b from n3 to n1 in graph1
added edge b from n3 to n2 in graph1
added edge b from n3 to n3 in graph1

# list nodes of graph
graph1 contains: n1 n2 n3

# list children of each node
the children of n1 in graph1 are: n1(a) n1(b) n1(c) n2(a) n2(b) n2(c) n3(a) n3(b) n3(c)
the children of n2 in graph1 are: n1(a) n1(b) n1(c) n2(a) n2(b) n2(c) n3(a) n3(b) n3(c)
the children of n3 in graph1 are: n1(a) n1(b) n1(c) n2(a) n2(b) n2(c) n3(a) n3(b) n3(c)